Expand AI Assistant query advice (#4294)
elastic/security-docs: docs/assistant/security-assistant.asciidoc

. Turn on the *Knowledge Base* option.
. Click *Save*. The knowledge base is now active.
+
When the knowledge base is active, a quick prompt for {esql} queries becomes available. It provides a good starting point for your {esql} conversations and questions.

[discrete]
[[ai-assistant-queries]]

> ### How to get the most from your queries

Reviewer: Some suggestions to make the heading slightly more concise:
Optimize your prompts
Optimize AI Assistant responses

Reviewer 2: +1 @natasha-moore-elastic's suggestions, or even just "Get the most from your queries" if you want to keep it conversational; this at least makes it more consistent with Elastic's general active-verb-first format for headings.

> The goal of the AI Assistant is to help users take full advantage of the Elastic Security platform to improve their cyber operations. Its ability to assist you depends on the specificity and detail of your questions. The more context and detail you provide, the more tailored and useful its responses will be. 

Reviewer 2: ```suggestion
Elastic AI Assistant can help users take full advantage of the {elastic-sec} platform to improve their cyber operations. Its ability to assist you depends on the specificity and detail of your questions. The more context and detail you provide, the more tailored and useful its responses will be. 
```

> To maximize its usefulness, consider using more detailed or elaborate prompts. For instance, after asking for an {esql} query example, you could ask a follow-up question like, “Could you give me some other examples and provide comments explaining the query you just gave?”. You can also ask for clarification or further exposition. 

Reviewer 2: Nit: "Elaborate" can have a slightly negative connotation; I think just "more detailed" is enough to make this point.

Also the example prompts here don't really include _more details_ in the prompts themselves; they're just asking for additional information. They're also asking for two different kinds of info (more examples, and explain the original example), so it'd help to mention each one separately.

```suggestion
To maximize its usefulness, consider using more detailed prompts or asking for additional information. For instance, after asking for an {esql} query example, you could ask a follow-up question like, “Could you give me some other examples?” You can also ask for clarification or further exposition, such as "Could you provide comments explaining the query you just gave?"
```

> To maximize its usefulness, consider using more detailed prompts or asking for additional information. For instance, after asking for an {esql} query example, you could ask a follow-up question like, “Could you give me some other examples?”. You can also ask for clarification or further exposition, for example "Please provide comments explaining the query you just gave".

Reviewer 2: Ending punctuation needs to go inside quotes, and a question mark is the only end punctuation needed:
```suggestion
To maximize its usefulness, consider using more detailed prompts or asking for additional information. For instance, after asking for an {esql} query example, you could ask a follow-up question like, “Could you give me some other examples?” You can also ask for clarification or further exposition, for example "Please provide comments explaining the query you just gave."
```

In addition to practical advice, AI Assistant can offer conceptual advice, tips, and best practices for enhancing your security measures. You can ask it, for example:

> . “How do I set up a machine learning job in Elastic Security to detect anomalies in network traffic volume over time?”
> . “I need to monitor for unusual file creation patterns that could indicate ransomware activity. How would I construct this query using EQL?”

Reviewer 2: A bulleted list is more appropriate here, because these aren't steps being performed in a certain order.
```suggestion
* “How do I set up a machine learning job in Elastic Security to detect anomalies in network traffic volume over time?”
* “I need to monitor for unusual file creation patterns that could indicate ransomware activity. How would I construct this query using EQL?”
```
